chain-impl-mockchain/tests/common/arbitrary/address.rs

```diff
@@ -0,0 +1,38 @@
+use crate::common::address::AddressData;
+use chain_addr::{Discrimination, KindType};
+use quickcheck::{Arbitrary, Gen};
+
+#[derive(Clone, Debug)]
+pub struct ArbitraryAddressKind(pub KindType);
+
+impl Arbitrary for ArbitraryAddressKind {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        match u8::arbitrary(g) % 3 {
+            0 => ArbitraryAddressKind(KindType::Single),
+            1 => ArbitraryAddressKind(KindType::Group),
+            2 => ArbitraryAddressKind(KindType::Account),
+            _ => unreachable!(),
```

Comment:
So what's the strategy for when a new `KindType` (ugh) appears? The test could silently fall into not producing really arbitrary values.

I would back this with an exhaustive match of the generated value, to make sure this impl is updated when the enum definition changes. If `Multisig` can't be properly handled at the moment, this impl can just ignore it and throw the dice again.

Reply:
Hmm but impl of  `Arbitrary for ArbitraryAddressKind` should handle situation with `Multisig` or code which is using it?

Reply:
I think you mentioned that the library itself currently panics on seeing `Multisig` in actual use, or something like that. If such is the case, this implementation, to be future-proof, needs to:
1. roll a random number in mod 4;
2. match it to all four values of `KindType`;
3. match the resulting enum value with an exhaustive set of patterns:

```rust
    use KindType::*;
    match v {
         Single | Group | Account => v,
         Multisig => {
             // Currently not supported by the library
             ... // use one of the above, or have a loop around the whole thing
                 // and use the continue statement here
         }
    }
```